crates/ra_project_model/src/json_project.rs

```diff
@@ -19,6 +19,8 @@ pub struct Crate {
     pub(crate) root_module: PathBuf,
     pub(crate) edition: Edition,
     pub(crate) deps: Vec<Dep>,
+    #[serde(default)]
+    pub(crate) features: Vec<String>,
 }
```

Comment:
`json_project` is intended to be independent from Cargo. So, because features are a strictly Cargo concept, we should expose raw cfgs and not features

Reply:
To be honest, I don't quite know how json project works. Is the json written manually?

Reply:
And is this `#[serde(default)]` really needed (for compatibility)?

Reply:
So, `json_project` exists for folks like Facebook, who don't use Cargo and build rust using another build system (like Buck). For such use-cases, I think it makes more sense to directly express the project model in terms of crates and `cfg`, and not in terms of higher-level Cargo concepts like packages and features. Ideally, `project.json` should be automatically generated from other build system config, but writing it manually is also an option. To be honest, I don't think anyone is really using this feature at present, but I think it's still important to maintain it, because it verifies that we don't depend on Cargo too much. 

And it's totally OK to not maintain any backwards compatability here, so I'll be fine with removing `#[serde(default)]`.

Reply:
Should it be `cfg: FxHashMap<String, Option<String>>`? Or two separated fields?
The first one is written as `{ "cfg": { "foo": "bar", "test": null } }` in json, which seems awkward.

Reply:
I think two separate fields would work better